src/lib/appLib.ts

```diff
@@ -0,0 +1,220 @@
+import { prodData, viewParam, viewCategoryArr, viewBrandArr } from '../globalTypes';
+
+class AppLib {
+    public products: prodData[];
+    public productsView: prodData[];
+    public productsInCart: prodData[];
+    public cartPriceTotal: number;
+    public cartAmountTotal: number;
+    public categoryArr: viewCategoryArr[];
+    public brandArr: viewBrandArr[];
+    public priceArr: number[];
+    public stockArr: number[];
+    public minPrice: number;
+    public maxPrice: number;
+    public minStock: number;
+    public maxStock: number;
+
+    constructor(productsData: prodData[]) {
+        this.products = productsData;
+        this.productsView = [];
+        this.productsInCart = [];
+        this.cartPriceTotal = 0;
+        this.cartAmountTotal = 0;
+        this.categoryArr = [];
+        this.brandArr = [];
+        this.priceArr = [];
+        this.stockArr = [];
+        this.minPrice = 0;
+        this.maxPrice = 0;
+        this.minStock = 0;
+        this.maxStock = 0;
+    }
+
+    getProductsData() {
+        return this.products;
```

Comment:
we can use arrow functions. And it will be really short:
getProductsData = () => this.products.